eZ/Publish/API/Repository/Values/ContentType/FieldDefinitionCollection.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\API\Repository\Values\ContentType;
+
+use ArrayAccess;
+use Closure;
+use Countable;
+use IteratorAggregate;
+
+interface FieldDefinitionCollection extends Countable, IteratorAggregate, ArrayAccess
+{
+    /**
+     * This method returns the field definition for the given identifier.
+     *
+     * @param string $fieldDefinitionIdentifier
+     *
+     * @return \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition|null
+     */
+    public function get(string $fieldDefinitionIdentifier): ?FieldDefinition;
+
+    /**
+     * This method returns true if the field definition for the given identifier exists.
+     *
+     * @param string $fieldDefinitionIdentifier
+     *
+     * @return bool
+     */
+    public function has(string $fieldDefinitionIdentifier): bool;
+
+    /**
+     * Return first element of collection.
+     *
+     * @return \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition|null
+     */
+    public function first(): ?FieldDefinition;
+
+    /**
+     * Return last element of collection.
+     *
+     * @return \eZ\Publish\API\Repository\Values\ContentType\FieldDefinition|null
+     */
+    public function last(): ?FieldDefinition;
+
+    /**
+     * Checks whether the collection is empty (contains no elements).
+     *
+     * @return bool TRUE if the collection is empty, FALSE otherwise.
+     */
+    public function isEmpty(): bool;
+
+    /**
+     * Returns all the elements of this collection that satisfy the predicate p.
+     * The order of the elements is preserved.
+     *
+     * @param Closure $p The predicate used for filtering.
+     *
+     * @return FieldDefinitionCollection A collection with the results of the filter operation.
+     */
+    public function filter(Closure $p): FieldDefinitionCollection;
+
+    /**
+     * Returns field definitions with given field type identifier.
+     *
+     * @param string $fieldTypeIdentifier
+     *
+     * @return FieldDefinitionCollection A collection with the results of the filter operation.
```

Comment:
inconsistent PHPDoc: some param and return types are FQCN, some are not. E.g.: ^